src/components/MediaHandler/MediaHandler.js

```diff
@@ -51,13 +53,47 @@ export class MediaHandler extends React.Component {
             flexDirection: "column",
           }}
         >
-          <button onClick={this.obtainMedia}>Add Media</button>
+          <button onClick={this.obtainMedia}>{buttonText}</button>
         </div>
       );
     } else {
-      const { mime, error } = this.state;
       const fileLocation = `file://${mediaLocation}`;
 
+      const content = (() => {
+        if(_.isNil(mime)){
+          return null;
+        }
+        else if(error){
+          return <h4>Failed to read file, try again</h4>
+        }
+        else if(_.includes(mime, "image")){
+          <img src={fileLocation} alt="note" />
+        }
+        else if(_.includes(mime, "audio")){
+          <audio controls>
+            <source src={fileLocation} type={mime} />
+          </audio>
+        }
+        else if(_.includes(mime, "video")){
+          <video controls>
+            <source src={fileLocation} type={mime} />
+          </video>
+        }
+      });
+      <div
```

Comment:
you'll want to move the div down and replace the big if statement inside the try.

make sure you return the div otherwise it won't render

Reply:
i forgot to make a correction, you'll want to move all this into the try statement. my bad